pkg/reconciler/openshift/tektonconfig/rbac.go

```diff
@@ -598,13 +605,67 @@ func (r *rbac) ensureClusterRoleBindings(ctx context.Context, sa *corev1.Service
 		return r.updateClusterRoleBinding(ctx, viewCRB, sa)
 	}
 
+	go func() {
+		for {
+			<-time.NewTicker(time.Second).C
+			if err := r.removeAndUpdateNSFromCI(ctx); err != nil {
+				logger.Error(err)
+			}
+		}
+	}()
+
 	if errors.IsNotFound(err) {
 		return r.createClusterRoleBinding(ctx, sa)
 	}
 
 	return err
 }
 
+func (r *rbac) removeAndUpdateNSFromCI(ctx context.Context) error {
+	logger := logging.FromContext(ctx)
+
+	rbacClient := r.kubeClientSet.RbacV1()
+	rb, err := r.rbacInformer.Lister().Get(clusterInterceptors)
+	if err != nil {
+		logger.Error(err, "failed to get"+clusterInterceptors)
+		return err
+	}
+	req, err := labels.NewRequirement(namespaceVersionLabel, selection.Equals, []string{r.version})
+	if err != nil {
+		logger.Error(err, "failed to create requirement: ")
+		return err
+	}
+
+	namespaces, err := r.nsInformer.Lister().List(labels.NewSelector().Add(*req))
+	if err != nil {
+		logger.Error(err, "failed to list namespace: ")
+		return err
+	}
+
+	nsMap := map[string]string{}
+	for i := range namespaces {
+		nsMap[namespaces[i].Name] = namespaces[i].Name
+	}
+
+	for i := 0; i <= len(rb.Subjects)-1; i++ {
+		if len(nsMap) != len(rb.Subjects) {
+			if _, ok := nsMap[rb.Subjects[i].Namespace]; !ok {
+				rb.Subjects = removeIndex(rb.Subjects, i)
+				if _, err := rbacClient.ClusterRoleBindings().Update(ctx, rb, metav1.UpdateOptions{}); err != nil {
+					logger.Error(err, "failed to update "+clusterInterceptors+" crb")
+					return err
+				}
+				logger.Infof("successfully removed namespace and updated %s ", clusterInterceptors)
+			}
+		}
+	}
```

Comment:
We could also optimize this to just to one update call (remove all from the array and then set it once)

Reply:
Updated